source/simulator/helper_functions.cc

```diff
@@ -2403,6 +2404,235 @@ namespace aspect
       }
     return new_linear_stokes_solver_tolerance;
   }
+
+
+
+  template <int dim>
+  void
+  Simulator<dim>::compute_parameter_update ()
+  {
+
+    // interior term
+    system_matrix = 0.;
+    system_rhs = 0.;
+
+    const unsigned int quadrature_degree = introspection.polynomial_degree.compositional_fields + 1;
+
+    // Gauss quadrature in the interior for best accuracy.
+    const QGauss<dim> quadrature_formula(quadrature_degree);
+    const unsigned int dofs_per_cell = finite_element.dofs_per_cell;
+    const unsigned int n_q_points = quadrature_formula.size();
+
+    const QGauss<dim-1> quadrature_formula_face(quadrature_degree);
+    const unsigned int n_q_face_points = quadrature_formula_face.size();
+
+    FEValues<dim> fe_values (*mapping,
+                             finite_element,
+                             quadrature_formula,
+                             update_values |
+                             update_gradients |
+                             update_quadrature_points |
+                             update_JxW_values);
+
+    FEValues<dim> fe_values_adjoint (*mapping,
```

Comment:
You dont use `fe_values_adjoint`, but even if you did, it is set up exactly like `fe_values` so you may as well use that one instead. If you use `fe_values` with the adjoint solution vector it will work just like `fe_values_adjoint` (a single FEValues object can be used on multiple solution vectors, as long as the solution vectors have the same structure).